src/metabase/driver/generic_sql/query_processor.clj

```diff
@@ -77,6 +77,22 @@
         (isa? special-type :type/UNIXTimestampMilliseconds) (sql/unix-timestamp->timestamp (driver) field :milliseconds)
         :else                                               field)))
 
+  SQLExpression
+  (formatted [{:keys [expr]}]
+    (hsql/raw expr))
```

Comment:
I've been thinking about this SQLExpression stuff. I think this might be expanding the surface area of our API by quite a bit. Would probably be worth talking through some of the implications of this.